src/hash.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright 2021-2022 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or MIT.
+
+use std::hash::BuildHasherDefault;
+
+#[derive(Default)]
+pub struct IdentityKeyHash(u64);
+pub type IdentityBuildHasher = BuildHasherDefault<IdentityKeyHash>;
+
+impl std::hash::Hasher for IdentityKeyHash {
+	fn finish(&self) -> u64 {
+		self.0
+	}
+
```

Comment:
you did not like my unreachables :) ? I guess it was a safe guard to avoid someone using the hasher incorrectly, but it takes loc, I am fine with removing them (will panic anyway).